uncertainties/core.py

```diff
@@ -65,139 +66,151 @@
 
 try:
     import numpy
+
+    __all__ += [
+        "correlated_values",
+        "correlated_values_norm",
+        "correlation_matrix",
+    ]
 except ImportError:
-    pass
-else:
-    # Entering variables as a block of correlated values.  Only available
-    # if NumPy is installed.
+    msg = "Unable to import numpy. Some functionality will be unavailable"
+    warnings.warn(msg)
```

Comment:
If I were using Uncertainties with Numpy, this warning every time I did `import uncertainties` would annoy me. It is better to defer the warning until the first attempt to use an unsupported feature (at which point it will need to be an exception instead of a warning).